pogam/scrapers/seloger.py

```diff
@@ -41,7 +41,18 @@
     )""",
     re.UNICODE | re.VERBOSE,
 )
-
+def MapCPtoCI (cp):
+    """ convert CP to CI as seloger use CI by default for search."""
+    try:
+        cp=str(cp)
+        df = pd.read_csv('https://www.data.gouv.fr/en/datasets/r/6d3428b2-3893-45a1-b404-2522a4e77d41',sep=';')
+        n=df[df['Code Postal'] == cp]['Code INSEE'].values[0]
+        n=str(n)
+        #get seloger readable ci with additional 0
+        n=n[0:2]+'0'+n[2:]
+        return n
+    except:
```

Comment:
This will  catch every single exception. For instance, if the file goes away, your `pd.read_csv` will error out, but will be interpret by your script as wrong post code input. It makes debugging more difficult (program is telling you "wrong post code", when the real error is missing file)

Best practices:
  - catch the specific exception(s) you anticipate
  - put as few lines of code as possible in the `try` block. 

In this case it looks like if the user supplies a wrong post code, `df[df['Code Postal'] == cp]['Code INSEE'].values` will be empty, so the `[0]` will raise an IndexError. So something like: 

```python
...
df = pd.read_csv(...)
try:
    n=df[df['Code Postal'] == cp]['Code INSEE'].values[0]
except IndexError:
    raise ValueError(f"Unknown post code f{cp}".)
n = str(n)
...
```